Extension/CloudFoundryCLI/cloudfoundrycli.ts

```diff
@@ -65,10 +65,16 @@ if (!cfPath) {
             if (args) {
                 cfCmd.line(args);
             }
-            cfCmd.exec()
-                .fail(function (err) {
-                    tl.setResult(tl.TaskResult.Failed, '' + err);
-                });
+            
+            var cfResult = cfCmd.execSync();
+            
+            if(cfResult.error != null){
+                tl.setResult(tl.TaskResult.Failed, '' + cfResult.error);
+            }
+            else{
+                tl.setVariable("CfCliOutput", cfResult.stdout, false);
```

Comment:
Variables should be small actionable things we set for the next steps or pipeline to use. The entire stdout from a command is excessive. Generally variables are say a version that was deployed or the path to deployment file. The variables are stored in the system environment and the entire output might be fit in the limits different OSes have as well.

Reply:
but you don't know what the command is so you can't get the right env variable. I added this so if use the CLI to query something you can actually use its output. do you have any other ideas how to do that then?

Reply:
I am not sure I follow the scenario, can you provide more detail? Wouldn't parsing the cfCommand input help with figuring out which command is being run. What data is of interest? Can you parse the stdout and avoid setting the variable if what you are looking for is not present?